### PR TITLE
Consistent import from top-level and first-level namespace of qcodes

### PR DESCRIPTION
**Pull request by lakhotiaharshit**
This pr removes does not import the legacy api in the top-level namespace of qcodes  by default. If the flag for importing legacy api is set to true in the config file then only the legacy-imports are done. It fixes the example notebooks to reflect this change. 

It also copies the imports to the __init__.py files of the respective modules. 

@jenshnielsen

### CODE REVIEW

File: qcodes/__init__.py [hunk 1]
```diff
@@ -38,11 +37,12 @@
               'try "from qcodes.plots.qcmatplotlib import MatPlot" '
               'to see the full error')
 
+if config.core.import_legacy_api:
+    from qcodes.loops import Loop, active_loop, active_data_set
+    from qcodes.measure import Measure
+    from qcodes.actions import Task, Wait, BreakIf
```
Comment (jenshnielsen) [inline qcodes/__init__.py]: I think these lines 

```
from qcodes.data.data_set import DataSet, new_data, load_data
from qcodes.data.location import FormatLocation
from qcodes.data.location import FormatLocation
from qcodes.data.data_array import DataArray
from qcodes.data.data_array import DataArray
from qcodes.data.format import Formatter
from qcodes.data.format import Formatter
from qcodes.data.gnuplot_format import GNUPlotFormat
from qcodes.data.gnuplot_format import GNUPlotFormat
from qcodes.data.hdf5_format import HDF5Format
from qcodes.data.hdf5_format import HDF5Format
from qcodes.data.io import DiskIO
from qcodes.data.io import DiskIO
```
should also be put behind the legacy flag

**Comment by jenshnielsen** [comment]
Looks good. Other than the change mentioned above I think this just need a changelog entry.

**Comment by jenshnielsen** [comment]
The travis failure probably points to a circular import. We may have to rething the importing slightly

File: qcodes/dataset/descriptions/versioning/serialization.py [hunk 2]
```diff
@@ -25,10 +25,9 @@
 from typing import Any, Dict, Type, Union
 
 from qcodes.utils.helpers import YAML
-import qcodes.dataset.descriptions.rundescriber as current
-import qcodes.dataset.descriptions.versioning.v0 as v0
-from qcodes.dataset.descriptions.versioning.converters import (
-    v0_to_v1, v1_to_v0)
+from ...descriptions import rundescriber as current
+from ...descriptions.versioning import v0 as v0
+from ...descriptions.versioning.converters import (v0_to_v1, v1_to_v0)
```
Comment (jenshnielsen) [inline qcodes/dataset/descriptions/versioning/serialization.py]: I would have expected 

```
from .. import rundescriber as current
from . import v0
from .converters import v0_to_v1, v1_to_v0
```
to be enought

File: qcodes/data/__init__.py [hunk 3]
```diff
@@ -1,7 +0,0 @@
-from qcodes.data.data_set import DataSet, new_data, load_data
-from qcodes.data.location import FormatLocation
-from qcodes.data.data_array import DataArray
-from qcodes.data.format import Formatter
-from qcodes.data.gnuplot_format import GNUPlotFormat
-from qcodes.data.hdf5_format import HDF5Format
-from qcodes.data.io import DiskIO
```
Comment (jenshnielsen) [inline qcodes/data/__init__.py]: Why delete these? I think they are fine to stay

File: qcodes/instrument_drivers/tektronix/Keithley_2600_channels.py [hunk 4]
```diff
@@ -5,7 +5,8 @@
 from typing import List, Dict, Optional
 
 import qcodes as qc
-from qcodes import VisaInstrument, DataSet
+from qcodes import VisaInstrument
+from qcodes.data.data_set import DataSet
```
Comment (jenshnielsen) [inline qcodes/instrument_drivers/tektronix/Keithley_2600_channels.py]: Internally this driver also calls qcodes.measure that should probably be fixed too

Comment (lakhotiaharshit) [reply qcodes/instrument_drivers/tektronix/Keithley_2600_channels.py]: Fixed at one place internally.

File: qcodes/data/__init__.py (hunk 3 above)
Comment (lakhotiaharshit) [reply qcodes/data/__init__.py]: reverted back

**Comment by jenshnielsen** [comment]
@lakhotiaharshit I think we just need to fix the 2 long lines then this should be ready to go

**Comment by lakhotiaharshit** [comment]
> @lakhotiaharshit I think we just need to fix the 2 long lines then this should be ready to go

@jenshnielsen '2 long lines' as in changelog? I just did that.

File: qcodes/data/__init__.py [hunk 5]
```diff
@@ -0,0 +1,6 @@
+from qcodes.data.location import FormatLocation
+from qcodes.data.data_array import DataArray
+from qcodes.data.format import Formatter
+from qcodes.data.gnuplot_format import GNUPlotFormat
+from qcodes.data.hdf5_format import HDF5Format
+from qcodes.data.io import DiskIO
```
Comment (astafan8) [inline qcodes/data/__init__.py]: shouldn't these be relative imports in order to make the "module" independent on where it itself is located?

File: qcodes/dataset/__init__.py [hunk 6]
```diff
@@ -2,6 +2,14 @@
...
 and from disk
 """
+from qcodes.dataset.measurements import Measurement
+from qcodes.dataset.data_set import new_data_set, load_by_counter, load_by_id, load_by_run_spec, load_by_guid
+from qcodes.dataset.experiment_container import new_experiment, load_experiment, load_experiment_by_name, \
+    load_last_experiment, experiments, load_or_create_experiment
+from qcodes.dataset.sqlite.settings import SQLiteSettings
+from qcodes.dataset.descriptions.param_spec import ParamSpec
+from qcodes.dataset.sqlite.database import initialise_database, \
+    initialise_or_create_database_at
```
Comment (astafan8) [inline qcodes/dataset/__init__.py]: and these perhaps should be relative?

File: qcodes/instrument/__init__.py [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    Parameter,
+    ArrayParameter,
+    MultiParameter,
+    ParameterWithSetpoints,
+    DelegateParameter,
+    ManualParameter,
+    ScaledParameter,
+    combine,
+    CombinedParameter)
+from qcodes.instrument.sweep_values import SweepFixedValues, SweepValues
```
Comment (astafan8) [inline qcodes/instrument/__init__.py]: and these should be relative as well?

**Comment by lakhotiaharshit** [comment]
> great work! just one question - shouldn't all sub-module-level imports be relative? (see other comments)

Yes, now that you point it out. I think they all should be relative.

File: qcodes/data/__init__.py (hunk 5 above)
Comment (lakhotiaharshit) [reply qcodes/data/__init__.py]: fixed

File: qcodes/dataset/__init__.py (hunk 6 above)
Comment (lakhotiaharshit) [reply qcodes/dataset/__init__.py]: fixed

File: qcodes/instrument/__init__.py (hunk 7 above)
Comment (lakhotiaharshit) [reply qcodes/instrument/__init__.py]: fixed

**Comment by jenshnielsen** [comment]
@lakhotiaharshit as in the feedback from codacy :)

**Pull request merged**